internal/cli/plugin/plugin.go

```diff
@@ -69,12 +82,14 @@ func (opts *Opts) findPluginWithName(name string) (*plugin.Plugin, error) {
 }
 
 func RegisterCommands(rootCmd *cobra.Command) {
-	plugins := plugin.GetAllValidPlugins(rootCmd.Commands())
+	plugins := plugin.GetAllValidPlugins(createExistingCommandsMap(rootCmd.Commands()))
```

Comment:
It would probably make more sense here to either:
- keep passing cobra commands
- pass a slice with all commands

See other remark for more details.